examples/pubsub/README_pubsub_tpm_keystore.txt

```diff
@@ -0,0 +1,103 @@
+1. TESTED ENVIRONMENT:
+Apollo Lake processor-1.60GHz
+OS - Ubuntu - 20.04.2 LTS
+Kernel - 5.4.0-74-generic
+
+2. ENVIRONMENT SETUP:
+Follow the below steps in both Publisher and Subscriber nodes.
+Install required packages:
+    sudo apt -y install acl autoconf autoconf-archive automake build-essential cmake doxygen gcc git iproute2 libcurl4-openssl-dev libjson-c-dev libcmocka0 libcmocka-dev libglib2.0-dev libini-config-dev libmbedtls-dev libssl-dev libsqlite3-dev libtool libyaml-dev pkg-config procps python3-pip sqlite3 udev uthash-dev
+
+Install the TPM2 Software Stack library:
+    cd ${HOME}
+    git clone https://github.com/tpm2-software/tpm2-tss.git
+    cd ${HOME}/tpm2-tss
+    ./bootstrap
+    ./configure --with-udevrulesdir=/etc/udev/rules.d --with-udevrulesprefix=70-
+    make -j$(nproc)
+    sudo make install
+    sudo ldconfig
+    sudo udevadm control --reload-rules && sudo udevadm trigger
+
+Install the TPM2 Tools (tpm2-tools):
+    sudo apt install tpm2-tools
+    sudo apt install opensc
+
+Install adaptation of PKCS#11 for TPM2 (tpm2-pkcs11):
+    cd ${HOME}
+    git clone https://github.com/tpm2-software/tpm2-pkcs11.git
+    cd ${HOME}/tpm2-pkcs11
+    ./bootstrap
+    ./configure
+    make -j$(nproc)
+    sudo make install
+    sudo ldconfig
+    cp ${HOME}/tpm2-pkcs11/src/pkcs11.h /usr/include
+
+Install the required python packages:
+    cd ${HOME}/tpm2-pkcs11/tools/
+    sudo pip3 install pyasn1_modules
+    pip3 install .
+
+Create an directory to create slot
+    cd ${HOME}/
+    mkdir pkcs11_store
+
+Set environment variable for TCTI connection and path for database:
+    cd ${HOME}/
+    echo 'export TPM2TOOLS_TCTI="device:/dev/tpm0"' >> .bashrc
+    echo 'export TPM2_PKCS11_STORE="'${HOME}'/pkcs11_store"' >> .bashrc
+    echo 'PATH=$PATH:'${HOME}'/pkcs11_store:'${HOME}'/tpm2-pkcs11/tools' >> .bashrc
+    source .bashrc
+
+creates the tpm2_pkcs11.sqlite3 database
+    sudo ${HOME}/tpm2-pkcs11/tools/tpm2_ptool.py init --path=/${HOME}/pkcs11_store
```

Comment:
`--path` can be omitted in all instances when `TPM2_PKCS11_STORE` env variable is set.